subsys/bluetooth/host/gatt.c

```diff
@@ -1019,11 +1021,16 @@ static u8_t gatt_foreach_iter(const struct bt_gatt_attr *attr,
 		return BT_GATT_ITER_CONTINUE;
 	}
 
-	if (*num_matches) {
+	if (num_matches) {
```

Comment:
Since we are not using num_matches as an optional pointer for "unlimited" but instead only give UINT16_MAX, maybe we could skip checking for NULL pointer in this function?

Reply:
Done.